src/biogeophys/SnowCoverFractionMod.F90

```diff
@@ -0,0 +1,785 @@
+       snow_depth, frac_sno)
+    !
+    ! !DESCRIPTION:
+    ! Update snow depth and snow fraction using the CLM5 parameterization
+    !
+    ! !ARGUMENTS:
+    class(snow_cover_fraction_clm5_type), intent(in) :: this
+    type(bounds_type), intent(in) :: bounds
+    integer, intent(in) :: num_c       ! number of columns in filter_c
+    integer, intent(in) :: filter_c(:) ! column filter to operate over
+
+    logical  , intent(in)    :: urbpoi( bounds%begc: )       ! true if the given column is urban
+    real(r8) , intent(in)    :: h2osno_total( bounds%begc: ) ! total snow water (mm H2O)
+    real(r8) , intent(in)    :: snowmelt( bounds%begc: )     ! total snow melt in the time step (mm H2O)
+    real(r8) , intent(in)    :: int_snow( bounds%begc: )     ! integrated snowfall (mm H2O)
+    real(r8) , intent(in)    :: newsnow( bounds%begc: )      ! total new snow in the time step (mm H2O)
+    real(r8) , intent(in)    :: bifall( bounds%begc: )       ! bulk density of newly fallen dry snow (kg/m3)
+
+    real(r8) , intent(inout) :: snow_depth( bounds%begc: )   ! snow height (m)
+    real(r8) , intent(inout) :: frac_sno( bounds%begc: )     ! fraction of ground covered by snow (0 to 1)
+    !
+    ! !LOCAL VARIABLES:
+    integer  :: fc, c
+    real(r8) :: fsno_new
+    real(r8) :: fmelt
+    real(r8) :: z_avg                                 ! grid cell average snow depth
+
+    character(len=*), parameter :: subname = 'UpdateSnowDepthAndFracClm5'
+    !-----------------------------------------------------------------------
+
+    SHR_ASSERT_FL((ubound(urbpoi, 1) == bounds%endc), sourcefile, __LINE__)
+    SHR_ASSERT_FL((ubound(h2osno_total, 1) == bounds%endc), sourcefile, __LINE__)
+    SHR_ASSERT_FL((ubound(snowmelt, 1) == bounds%endc), sourcefile, __LINE__)
+    SHR_ASSERT_FL((ubound(int_snow, 1) == bounds%endc), sourcefile, __LINE__)
+    SHR_ASSERT_FL((ubound(newsnow, 1) == bounds%endc), sourcefile, __LINE__)
+    SHR_ASSERT_FL((ubound(bifall, 1) == bounds%endc), sourcefile, __LINE__)
+    SHR_ASSERT_FL((ubound(snow_depth, 1) == bounds%endc), sourcefile, __LINE__)
+    SHR_ASSERT_FL((ubound(frac_sno, 1) == bounds%endc), sourcefile, __LINE__)
+
+    do fc = 1, num_c
+       c = filter_c(fc)
+       if (h2osno_total(c) > 0.0_r8) then
+          !======================  FSCA PARAMETERIZATIONS  ======================
+          ! fsca parameterization based on *changes* in swe
+          ! first compute change from melt during previous time step
+          if(snowmelt(c) > 0._r8) then
+             frac_sno(c) = this%FracSnowDuringMelt( &
+                  c            = c, &
+                  h2osno_total = h2osno_total(c), &
+                  int_snow     = int_snow(c))
+          end if
+
+          ! update fsca by new snow event, add to previous fsca
+          if (newsnow(c) > 0._r8) then
+             fsno_new = 1._r8 - (1._r8 - tanh(this%accum_factor * newsnow(c))) * (1._r8 - frac_sno(c))
+             frac_sno(c) = fsno_new
```

Comment:
is there a benefit to keeeping this fsno_new variable?

Reply:
No benefit; I'll remove it: thanks.